tests/integration/specs/common/autostore.spec.js

```diff
@@ -0,0 +1,1147 @@
+      try {
+        await networkTypeCollection.findById(sampleDocs[0]._id);
+        throw new Error('This test should fail');
+      } catch (error) {
+        expect(error).to.be.instanceOf(Errors.NotFoundError);
+      }
+    });
+
+    it('should return 0 when no items are deleted', async function() {
+      const autoTypeCollection = DataStore.collection(collectionName, DataStoreType.Auto);
+      const syncTypeCollection = DataStore.collection(collectionName, DataStoreType.Sync);
+      const networkTypeCollection = DataStore.collection(collectionName, DataStoreType.Network);
+
+      // Create sample docs
+      const propertyName = 'title';
+      const sampleDocs = [];
+      sampleDocs.push(await autoTypeCollection.create({ [propertyName]: randomString() }));
+      sampleDocs.push(await autoTypeCollection.create({ [propertyName]: randomString() }));
+      sampleDocs.push(await autoTypeCollection.create({ [propertyName]: randomString() }));
+
+      // Create a query
+      const query = new Query()
+        .equalTo(propertyName, randomString());
+
+      // Remove docs
+      expect(await autoTypeCollection.remove(query)).to.deep.equal({ count: 0 });
+
+      // Find with Network
+      expect(await networkTypeCollection.find()).to.deep.equal(sampleDocs);
+
+      // Find with Sync
+      expect(await syncTypeCollection.find().toPromise()).to.deep.equal(sampleDocs);
+    });
+
+    it('should return an error for invalid query');
+
+    it('should remove item locally and create delete operation in the sync queue with connectivity error');
+
+    it('should delete locally stored items that are deleted from the backend', async function() {
+      const autoTypeCollection = DataStore.collection(collectionName, DataStoreType.Auto);
+      const syncTypeCollection = DataStore.collection(collectionName, DataStoreType.Sync);
+      const networkTypeCollection = DataStore.collection(collectionName, DataStoreType.Network);
+
+      // Create sample docs
+      const propertyName = 'title';
+      const sampleDocs = [];
+      sampleDocs.push(await autoTypeCollection.create({ [propertyName]: randomString() }));
+      sampleDocs.push(await autoTypeCollection.create({ [propertyName]: randomString() }));
+
+      // Create a query
+      const query = new Query()
+        .contains(propertyName, [sampleDocs[0][propertyName]]);
+
+      // Remove docs with Network
+      expect(await networkTypeCollection.remove(query)).to.deep.equal({ count: 1 });
+
+      // Remove docs with Auto
+      expect(await autoTypeCollection.remove(query)).to.deep.equal({ count: 0 });
+
+      // Find with Sync
+      try {
+        await syncTypeCollection.findById(sampleDocs[0]._id).toPromise();
+        throw new Error('This test should fail');
+      } catch (error) {
+        expect(error).to.be.instanceOf(Errors.NotFoundError);
+      }
+    });
+
+    it.skip('should delete items with tagged store', async function() {
+      const tag = randomString();
```

Comment:
Is it failing?

Reply:
https://github.com/Kinvey/js-sdk/pull/409/commits/04da399a69301965b65e950a33ae05133c91b959